prow/cmd/webhook-server/helpers.go

```diff
@@ -246,37 +254,183 @@ func createValidatingWebhookConfig(ctx context.Context, caPem string, client ctr
 		}
 	} else if err != nil {
-		return fmt.Errorf("failed to create validation webhook config: %w", err)
+		return fmt.Errorf("failed to create validating webhook config: %w", err)
 	}
 
 	return nil
 }
 
 func patchValidatingWebhookConfig(ctx context.Context, caPem string, client ctrlruntimeclient.Client) error {
 	key := types.NamespacedName{
-		Namespace: "default",
-		Name:      "prow-job-validating-webhook-config.prow.k8s.io",
+		Namespace: defaultNamespace,
+		Name:      prowJobValidatingWebhookName,
 	}
 
 	patchOptions := &ctrlruntimeclient.PatchOptions{
 		FieldManager: "webhook-server",
 	}
 	var validatingWebhookConfig admregistration.ValidatingWebhookConfiguration
 	if err := client.Get(ctx, key, &validatingWebhookConfig); err != nil {
-		return fmt.Errorf("failed to get validation webhook config: %w", err)
+		return fmt.Errorf("failed to get validating webhook config: %w", err)
 	}
 	oldValidatingWebhook := validatingWebhookConfig.DeepCopy()
 	validatingWebhookConfig.Webhooks[0].ClientConfig.CABundle = []byte(caPem)
 	if err := client.Patch(ctx, &validatingWebhookConfig, ctrlruntimeclient.MergeFrom(oldValidatingWebhook), patchOptions); err != nil {
-		return fmt.Errorf("failed to patch validation webhook config: %w", err)
+		return fmt.Errorf("failed to patch validating webhook config: %w", err)
+	}
+	return nil
+}
+
+func createMutatingWebhookConfig(ctx context.Context, caPem string, client ctrlruntimeclient.Client) error {
+	operations := []admregistration.OperationType{"CREATE"}
+	scope := admregistration.ScopeType("*")
+	path := "/mutate"
+	sideEffects := admregistration.SideEffectClass("None")
+
+	mutatingWebhookConfig := &admregistration.MutatingWebhookConfiguration{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "MutatingWebhookConfiguration",
+			APIVersion: "admissionregistration.k8s.io/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: prowJobMutatingWebhookName,
+		},
+		Webhooks: []admregistration.MutatingWebhook{
+			{
+				Name: prowJobMutatingWebhookName,
+				ObjectSelector: &v1.LabelSelector{
+					MatchLabels: map[string]string{
+						"admission-webhook": "enabled",
+						"default-me":        "enabled",
+					},
+				},
+				Rules: []admregistration.RuleWithOperations{
+					{
+						Operations: operations,
+						Rule: admregistration.Rule{
+							APIGroups:   []string{"prow.k8s.io"},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"prowjobs"},
+							Scope:       &scope,
+						},
+					},
+				},
+				ClientConfig: admregistration.WebhookClientConfig{
+					Service: &admregistration.ServiceReference{
+						Namespace: defaultNamespace,
+						Name:      prowjobAdmissionServiceName,
+						Path:      &path,
+					},
+					CABundle: []byte(caPem),
+				},
+				SideEffects:             &sideEffects,
+				AdmissionReviewVersions: []string{"v1"},
+			},
+		},
+	}
+
+	createOptions := &ctrlruntimeclient.CreateOptions{
+		FieldManager: "webhook-server",
```

Comment:
what does this do? could you add a comment? And if this is important please also make it a const

Reply:
not really important just indicates that the component that created, the configuration is the webhook server

Reply:
do you mean that it's not useful? or the name doesn't matter? if the former we should delete these 3 lines, otherwise please add a comment saying that it just needs to be set, and the value doesn't matter